### DISCUSSION TITLE
Execution failed for task ':capacitor-community-bluetooth-le:compileDebugKotlin'

### DISCUSSION THREAD

**Discussion by unstopp** [capacitor-community/bluetooth-le#536] [Q&A] [answered]
Hi everyone,

I am having a bit of a blocker when trying to run my Ionic project on a device (either simulator or physical device). The build is fine, it just happens when trying to run the app.

The following exception is thrown:

**org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':capacitor-community-bluetooth-le:compileDebugKotlin'.**

**Caused by: org.gradle.api.GradleException: 'compileDebugJavaWithJavac' task (current target is 11) and 'compileDebugKotlin' task (current target is 17) jvm target compatibility should be set to the same Java version.**
Consider using JVM toolchain: https://kotl.in/gradle/jvm/toolchain

Project is running on Ionic 7 and Capacitor 5. Bluetooth-le version is 3.0.0-1. Java version is 17 and simulator device is Pixel 5 API 31.
I have checked various build & compile settings that I could find in order to be the same api and/or the same java version.
Any feedback is much appreciated, thank you!

Best,
Marius

**Comment by pwespi** [comment, marked as answer]
This should be fixed in version `3.0.0` of the plugin.

**Reply by pwespi** [reply]
See https://github.com/capacitor-community/bluetooth-le/issues/529 and https://github.com/capacitor-community/bluetooth-le/issues/533

**Reply by unstopp** [reply]
I installed 3.0.0 (@latest) and it's working. Thank you!
